src/main/java/pl/psnc/dei/controllers/BatchController.java

```diff
@@ -117,36 +87,23 @@ public BatchController(BatchService batchService, ImportPackageService importSer
 	 */
 	@PostMapping(path = "/complex-imports", consumes = MediaType.MULTIPART_FORM_DATA_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
 	public ResponseEntity<List<Import>> splitImport(@RequestParam(value = "projectName") String projectName,
-											  		@RequestParam(value = "datasetName", required = false) String datasetName,
-											  		@RequestParam(value = "name", required = false) String name,
-											  		@RequestBody @RequestParam("file") MultipartFile file) throws IOException {
-		List<Set<String>> records = batchService.splitImport(file);
-		List<Import> imports = new ArrayList<>();
-
+													@RequestParam(value = "datasetName", required = false) String datasetName,
+													@RequestParam(value = "name", required = false) String name,
+													@RequestBody @RequestParam("file") MultipartFile file) throws IOException {
 		try {
-			int counter = 0;
-			String importName;
-
-			for(Set<String> identifiers : records) {
-				Set<Record> uploadedRecords = uploadRecordsToProject(projectName, datasetName, identifiers);
-				if (!uploadedRecords.isEmpty()) {
-					if (StringUtils.isBlank(name)) {
-						importName = name;
-					} else {
-						importName = name + "_" + counter++;
-					}
-					imports.add(importService.createImport(importName, uploadedRecords.iterator().next().getProject().getProjectId(), uploadedRecords));
-				}
-			}
+			return ResponseEntity.ok(this.batchService.makeComplexImport(file, name, projectName, datasetName));
 		} catch (NotFoundException e) {
+			e.printStackTrace();
 			return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+		} catch (Exception e) {
+			e.printStackTrace();
+			return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
```

Comment:
I'm not sure that catching all of the exceptions and sending response to user "this is your failure, refine you request" is good, because anytime an internal server error could happen, and server should take all of the responsibility by sending 5xx response.

Reply:
as there is probably no good on-stop solution, this catch was permanently removed